library/src/scripts/components/headers/vanillaHeaderStyles.ts

```diff
@@ -0,0 +1,555 @@
+    const formElementVars = formElementsVariables();
+    const userPhotoVars = userPhotoVariables();
+    const headerColors = vars.colors;
+    const vanillaMenuVars = vanillaMenuVariables();
+    const mediaQueries = layoutStyles().mediaQueries();
+    const flex = flexHelper();
+
+    const root = style(
+        {
+            backgroundColor: headerColors.bg.toString(),
+            color: headerColors.fg.toString(),
+            $nest: {
+                "&.isFixed": {
+                    position: "fixed",
+                    top: 0,
+                    left: 0,
+                    right: 0,
+                    zIndex: 1,
+                },
+                ".searchBar__control": {
+                    color: vars.colors.fg.toString(),
+                },
+                ".suggestedTextInput-clear.searchBar-clear": {
+                    color: vars.colors.fg.toString(),
+                    $nest: {
+                        "&:hover": {
+                            color: vars.colors.fg.toString(),
+                        },
+                        "&:active": {
+                            color: vars.colors.fg.toString(),
+                        },
+                        "&:focus": {
+                            color: vars.colors.fg.toString(),
+                        },
+                    },
+                },
+                ".searchBar__placeholder": {
+                    color: vars.colors.fg.fade(0.8).toString(),
+                },
+            },
+        },
+        mediaQueries.oneColumn({
+            height: px(vars.sizing.mobile.height),
+        }),
+    );
+
+    const spacer = style({ height: px(vars.sizing.height) });
+
+    const bar = style(
+        {
+            display: "flex",
+            justifyContent: "space-between",
+            flexWrap: "nowrap",
+            alignItems: "center",
+            height: px(vars.sizing.height),
+            width: percent(100),
+            $nest: {
+                "&.isHome": {
+                    justifyContent: "space-between",
+                },
+            },
+        },
+        mediaQueries.oneColumn({ height: px(vars.sizing.mobile.height) }),
+    );
+
+    const logoContainer = style(
+        {
+            display: "inline-flex",
+            alignSelf: "center",
+            flexBasis: vars.endElements.flexBasis,
+            color: vars.colors.fg.toString(),
+            $nest: {
+                "&.focus-visible": {
+                    $nest: {
+                        "&.headerLogo-logoFrame": {
+                            outline: `5px solid ${vars.buttonContents.hover.bg}`,
+                            background: vars.buttonContents.hover.bg.toString(),
+                            borderRadius: vars.button.borderRadius,
+                        },
+                    },
+                },
+            },
+        },
+        mediaQueries.oneColumn({ height: px(vars.sizing.mobile.height) }),
+    );
+
+    const meBox = style({
```

Comment:
Could we break this into a separate partial in the `components/mebox` directory? I feel like this partial is starting to get a little large.

Reply:
These are "vanillaHeader-meBox" styles, not just meBox styles